src/test/java/io/stargate/sgv2/jsonapi/fixtures/testdata/TestData.java

```diff
@@ -0,0 +1,55 @@
+package io.stargate.sgv2.jsonapi.fixtures.testdata;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TestData {
+
+  protected final TestDataNames names;
+
+  private final Map<Class<?>, Object> cache = new HashMap<>();
+
+  public TestData() {
+    this(new TestDataNames());
+  }
+
+  protected TestData(TestDataNames names) {
+    this.names = names;
+  }
+
+  @SuppressWarnings("unchecked")
+  private <T> T getOrCache(Class<T> clazz) {
+    return (T)
+        cache.computeIfAbsent(
+            clazz,
+            k -> {
+              try {
+                Constructor<?> constructor = clazz.getConstructor(TestData.class);
+                return constructor.newInstance(this);
+              } catch (NoSuchMethodException
```

Comment:
just `catch (Exception e)` and be done?

Reply:
will update shortly in another PR to use the wider catch